From afaee00d18cc658b0b2f1cf02e485711a5e7217e Mon Sep 17 00:00:00 2001
From: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
Date: Mon, 1 Jul 2024 16:27:22 +0530
Subject: [PATCH 3/4] overlay: t23x: adsd3500: Changed the CSI port index to 1
 and lane polarity with 6

The CAM0 J20 connector connects the CSI 1 Data 0/1 MIPI lane connection
to the hexagon tof adapter board

Signed-off-by: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
---
 .../tegra234-p3767-camera-p3768-adsd3500.dts  | 162 ++++++++++++++++--
 1 file changed, 151 insertions(+), 11 deletions(-)

diff --git a/overlay/tegra234-p3767-camera-p3768-adsd3500.dts b/overlay/tegra234-p3767-camera-p3768-adsd3500.dts
index 69fdc86..fec6696 100644
--- a/overlay/tegra234-p3767-camera-p3768-adsd3500.dts
+++ b/overlay/tegra234-p3767-camera-p3768-adsd3500.dts
@@ -24,7 +24,7 @@
 					vi_port0: port@0 {
 							reg = <0>;
 							adsd3500_vi_in0: endpoint {
-								port-index = <0>;
+								port-index = <1>;
 								bus-width = <2>;
 								remote-endpoint = <&adsd3500_csi_out0>;
 							};
@@ -101,7 +101,7 @@
 								csi_chan0_port0: port@0 {
 									reg = <0>;
 									adsd3500_csi_in0: endpoint@0 {
-										port-index = <0>;
+										port-index = <1>;
 										bus-width = <2>;
 										remote-endpoint = <&adsd3500_out0>;
 										};
@@ -229,7 +229,8 @@
 						mode0 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -261,7 +262,8 @@
 						mode1 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -293,7 +295,8 @@
 						mode2 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -325,7 +328,8 @@
 						mode3 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -357,7 +361,8 @@
 						mode4 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -389,7 +394,8 @@
 						mode5 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -421,7 +427,8 @@
 						mode6 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -453,7 +460,8 @@
 						mode7 {
 							mclk_khz = "104250";
 							num_lanes = "2";
-							tegra_sinterface = "serial_c";
+							tegra_sinterface = "serial_b";
+							lane_polarity = "6";
 							phy_mode = "DPHY";
 							discontinuous_clk = "yes";
 							dpcm_enable = "false";
@@ -482,6 +490,138 @@
 							max_exp_time = "660000";
 							embedded_metadata_height = "1";
 						};
+						mode8 {
+                                                        mclk_khz = "104250";
+                                                        num_lanes = "2";
+                                                        tegra_sinterface = "serial_b";
+							lane_polarity = "6";
+                                                        phy_mode = "DPHY";
+                                                        discontinuous_clk = "yes";
+                                                        dpcm_enable = "false";
+                                                        cil_settletime = "0";
+                                                        dynamic_pixel_bit_depth = "12";
+                                                        csi_pixel_bit_depth = "12";
+                                                        mode_type = "bayer";
+                                                        pixel_phase = "bggr";
+
+                                                        active_w = "1024";
+                                                        active_h = "1024";
+                                                        readout_orientation = "0";
+                                                        line_length = "1024";
+                                                        inherent_gain = "1";
+                                                        mclk_multiplier = "4";
+                                                        pix_clk_hz = "208333333";
+
+                                                        min_gain_val = "0";
+                                                        max_gain_val = "48";
+                                                        gain_step_pitch = "0.3";
+                                                        min_hdr_ratio = "1";
+                                                        max_hdr_ratio = "1";
+                                                        min_framerate = "1.5";
+                                                        max_framerate = "30";
+                                                        min_exp_time = "30";
+                                                        max_exp_time = "660000";
+                                                        embedded_metadata_height = "1";
+                                                };
+						mode9 {
+                                                        mclk_khz = "104250";
+                                                        num_lanes = "2";
+                                                        tegra_sinterface = "serial_b";
+                                                        lane_polarity = "6";
+                                                        phy_mode = "DPHY";
+                                                        discontinuous_clk = "yes";
+                                                        dpcm_enable = "false";
+                                                        cil_settletime = "0";
+                                                        dynamic_pixel_bit_depth = "8";
+                                                        csi_pixel_bit_depth = "8";
+                                                        mode_type = "bayer";
+                                                        pixel_phase = "bggr";
+
+                                                        active_w = "2048";
+                                                        active_h = "2560";
+                                                        readout_orientation = "0";
+                                                        line_length = "2048";
+                                                        inherent_gain = "1";
+                                                        mclk_multiplier = "4";
+                                                        pix_clk_hz = "312500000";
+
+                                                        min_gain_val = "0";
+                                                        max_gain_val = "48";
+                                                        gain_step_pitch = "0.3";
+                                                        min_hdr_ratio = "1";
+                                                        max_hdr_ratio = "1";
+                                                        min_framerate = "1.5";
+                                                        max_framerate = "30";
+                                                        min_exp_time = "30";
+                                                        max_exp_time = "660000";
+                                                        embedded_metadata_height = "1";
+                                                };
+						mode10 {
+                                                        mclk_khz = "104250";
+                                                        num_lanes = "2";
+                                                        tegra_sinterface = "serial_b";
+                                                        lane_polarity = "6";
+                                                        phy_mode = "DPHY";
+                                                        discontinuous_clk = "yes";
+                                                        dpcm_enable = "false";
+                                                        cil_settletime = "0";
+                                                        dynamic_pixel_bit_depth = "8";
+                                                        csi_pixel_bit_depth = "8";
+                                                        mode_type = "bayer";
+                                                        pixel_phase = "bggr";
+
+                                                        active_w = "2048";
+                                                        active_h = "3328";
+                                                        readout_orientation = "0";
+                                                        line_length = "2048";
+                                                        inherent_gain = "1";
+                                                        mclk_multiplier = "4";
+                                                        pix_clk_hz = "312500000";
+
+                                                        min_gain_val = "0";
+                                                        max_gain_val = "48";
+                                                        gain_step_pitch = "0.3";
+                                                        min_hdr_ratio = "1";
+                                                        max_hdr_ratio = "1";
+                                                        min_framerate = "1.5";
+                                                        max_framerate = "30";
+                                                        min_exp_time = "30";
+                                                        max_exp_time = "660000";
+                                                        embedded_metadata_height = "1";
+                                                };
+						mode11 {
+                                                        mclk_khz = "104250";
+                                                        num_lanes = "2";
+                                                        tegra_sinterface = "serial_b";
+                                                        lane_polarity = "6";
+                                                        phy_mode = "DPHY";
+                                                        discontinuous_clk = "yes";
+                                                        dpcm_enable = "false";
+                                                        cil_settletime = "0";
+                                                        dynamic_pixel_bit_depth = "12";
+                                                        csi_pixel_bit_depth = "12";
+                                                        mode_type = "bayer";
+                                                        pixel_phase = "bggr";
+
+                                                        active_w = "1024";
+                                                        active_h = "2304";
+                                                        readout_orientation = "0";
+                                                        line_length = "1024";
+                                                        inherent_gain = "1";
+                                                        mclk_multiplier = "4";
+                                                        pix_clk_hz = "208333333";
+
+                                                        min_gain_val = "0";
+                                                        max_gain_val = "48";
+                                                        gain_step_pitch = "0.3";
+                                                        min_hdr_ratio = "1";
+                                                        max_hdr_ratio = "1";
+                                                        min_framerate = "1.5";
+                                                        max_framerate = "30";
+                                                        min_exp_time = "30";
+                                                        max_exp_time = "660000";
+                                                        embedded_metadata_height = "1";
+                                                };
 						ports {
 							#address-cells = <1>;
 							#size-cells = <0>;
@@ -489,7 +629,7 @@
 								reg = <0>;
 								adsd3500_out0: endpoint {
 									status = "okay";
-									port-index = <0>;
+									port-index = <1>;
 									bus-width = <2>;
 									remote-endpoint = <&adsd3500_csi_in0>;
 								};
-- 
2.34.1

